runtime/queries/resource_watermark.go

```diff
@@ -114,6 +114,10 @@ func (q *ResourceWatermark) resolveMetricsView(ctx context.Context, rt *runtime.
 	}
 
 	if !t.IsZero() {
+		// Hacky workaround for the following issue: the watermark is used as the *exclusive* upper bound for time ranges.
+		// We need to add a small delta to ensure the row with the watermark is included in the result.
+		t = t.Add(time.Second)
```

Comment:
Would just a ms be enough?

Reply:
I tried it and it didn't work outright with ClickHouse – apparently they have timestamps at second granularity by default